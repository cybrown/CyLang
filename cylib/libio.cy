void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
