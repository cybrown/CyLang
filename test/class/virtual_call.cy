import libio;

class A {
	int bar;
	virtual void p1(int i) {
		print("I am an A\n");
		return;
	}
	virtual void p2(int i) {
		print("I've said I am an A !\n");
		return;
	}
}

class B extends A {
	int foo;

	override void p1(int i) {
		print("I am a B\n");
		return;
	}
	override void p2(int i) {
		print("I've said I am a B !\n");
		return;
	}
}

void exec_p(A param) {
	param.p1(3);
	param.p2(3);
}

A a = A();
B b = B();

exec_p(a);
exec_p(b);
