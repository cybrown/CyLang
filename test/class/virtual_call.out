I am an A
I've said I am an A !
I am a B
I've said I am a B !
